Error opening file with newest unstable

Opening attached ORK file gives the following error:
![image](https://user-images.githubusercontent.com/243872/81582741-60f43180-937e-11ea-9c97-6cc22d4ee770.png)

This cropped up recently; it works fine in the build from late April tagged 1c7171ef.

[blackfish_golden_sunset.ork.zip](https://github.com/openrocket/openrocket/files/4610684/blackfish_golden_sunset.ork.zip)

```
   > net.sf.openrocket.file.RocketLoadException: Exception loading stream: Index 2 out of bounds for length 2
   > 	at net.sf.openrocket.file.GeneralRocketLoader.load(GeneralRocketLoader.java:98)
   > 	at net.sf.openrocket.gui.util.OpenFileWorker.doInBackground(OpenFileWorker.java:72)
   > 	at net.sf.openrocket.gui.util.OpenFileWorker.doInBackground(OpenFileWorker.java:28)
   > 	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
   > 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
   > 	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
   > 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
   > 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
   > 	at java.base/java.lang.Thread.run(Thread.java:834)
   > Caused by: java.lang.IndexOutOfBoundsException: Index 2 out of bounds for length 2
   > 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
   > 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
   > 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
   > 	at java.base/java.util.Objects.checkIndex(Objects.java:372)
   > 	at java.base/java.util.ArrayList.get(ArrayList.java:458)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.getChild(RocketComponent.java:1498)
   > 	at net.sf.openrocket.rocketcomponent.SymmetricComponent.getPreviousSymmetricComponent(SymmetricComponent.java:587)
   > 	at net.sf.openrocket.rocketcomponent.Transition.getForeRadius(Transition.java:77)
   > 	at net.sf.openrocket.rocketcomponent.Transition.getRadius(Transition.java:356)
   > 	at net.sf.openrocket.rocketcomponent.FinSet.getFinFront(FinSet.java:938)
   > 	at net.sf.openrocket.rocketcomponent.FreeformFinSet.clampFirstPoint(FreeformFinSet.java:291)
   > 	at net.sf.openrocket.rocketcomponent.FreeformFinSet.update(FreeformFinSet.java:276)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.componentChanged(RocketComponent.java:291)
   > 	at net.sf.openrocket.rocketcomponent.FinSet.componentChanged(FinSet.java:798)
   > 	at net.sf.openrocket.rocketcomponent.Rocket.fireComponentChangeEvent(Rocket.java:460)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.fireComponentChangeEvent(RocketComponent.java:1761)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.fireComponentChangeEvent(RocketComponent.java:1773)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.fireAddRemoveEvent(RocketComponent.java:1485)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.addChild(RocketComponent.java:1398)
   > 	at net.sf.openrocket.rocketcomponent.RocketComponent.addChild(RocketComponent.java:1353)
   > 	at net.sf.openrocket.file.openrocket.importt.ComponentHandler.openElement(ComponentHandler.java:52)
   > 	at net.sf.openrocket.file.simplesax.DelegatorHandler.startElement(DelegatorHandler.java:53)
   > 	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:510)
   > 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:374)
   > 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2710)
   > 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
   > 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
   > 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534)
   > 	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
   > 	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
   > 	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
   > 	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216)
   > 	at net.sf.openrocket.file.simplesax.SimpleSAX.readXML(SimpleSAX.java:49)
   > 	at net.sf.openrocket.file.openrocket.importt.OpenRocketLoader.loadFromStream(OpenRocketLoader.java:50)
   > 	at net.sf.openrocket.file.AbstractRocketLoader.load(AbstractRocketLoader.java:21)
   > 	at net.sf.openrocket.file.GeneralRocketLoader.loadUsing(GeneralRocketLoader.java:240)
   > 	at net.sf.openrocket.file.GeneralRocketLoader.loadRocket(GeneralRocketLoader.java:201)
   > 	at net.sf.openrocket.file.GeneralRocketLoader.loadStep1(GeneralRocketLoader.java:160)
   > 	at net.sf.openrocket.file.GeneralRocketLoader.load(GeneralRocketLoader.java:94)
   > 	... 8 more
```

.... 

```
 > Caused by: java.lang.IndexOutOfBoundsException: Index 2 out of bounds for length 2
   . . . 

   > at net.sf.openrocket.rocketcomponent.SymmetricComponent.getPreviousSymmetricComponent(SymmetricComponent.java:587)
   > 	at net.sf.openrocket.rocketcomponent.Transition.getForeRadius(Transition.java:77)
   > 	at net.sf.openrocket.rocketcomponent.Transition.getRadius(Transition.java:356)
   > 	at net.sf.openrocket.rocketcomponent.FinSet.getFinFront(FinSet.java:938)
```

--- a/core/src/net/sf/openrocket/rocketcomponent/FinSet.java
+++ b/core/src/net/sf/openrocket/rocketcomponent/FinSet.java
@@ -299,7 +299,7 @@ public abstract class FinSet extends ExternalComponent implements RingInstanceab
 	/** 
 	 * internally, set the internal offset and optionally validate tab
 	 * 
-	 * @param offsetRequest new requested tob offset
+	 * @param offsetRequest new requested tab offset
 	 */
 	public void setTabOffset( final double offsetRequest) {
 		tabOffset = offsetRequest;

--- a/core/src/net/sf/openrocket/rocketcomponent/SymmetricComponent.java
+++ b/core/src/net/sf/openrocket/rocketcomponent/SymmetricComponent.java
@@ -608,21 +608,21 @@ public abstract class SymmetricComponent extends BodyComponent implements Radial
 			return null;
 		}
 
+		final ComponentAssembly assembly = this.getAssembly();
 		// might be: (a) Rocket -- for centerline stages
 		//           (b) BodyTube -- for Parallel Stages
-		final AxialStage stage = this.getStage();
-		final RocketComponent stageParent = stage.getParent();
+		final RocketComponent assemblyParent = assembly.getParent();
 
 		// note:  this is not guaranteed to _contain_ a stage... but that we're _searching_ for one.
-		int stageIndex = stageParent.getChildPosition(stage);
+		int assemblyIndex = assemblyParent.getChildPosition(assembly);
 		int symmetricIndex = this.parent.getChildPosition(this) + 1;
 
-		while(stageIndex < stageParent.getChildCount()) {
-			final RocketComponent nextStage = stageParent.getChild(stageIndex);
+		while(assemblyIndex < assemblyParent.getChildCount()) {
+			final RocketComponent searchAssembly = assemblyParent.getChild(assemblyIndex);
 
-			if(nextStage instanceof AxialStage){
-				while (symmetricIndex < nextStage.getChildCount()) {
-					final RocketComponent nextSymmetric = nextStage.getChild(symmetricIndex);
+			if(searchAssembly instanceof ComponentAssembly){
+				while (symmetricIndex < searchAssembly.getChildCount()) {
+					final RocketComponent nextSymmetric = searchAssembly.getChild(symmetricIndex);
 
 					if (nextSymmetric instanceof SymmetricComponent) {
 						return (SymmetricComponent) nextSymmetric;
@@ -630,8 +630,8 @@ public abstract class SymmetricComponent extends BodyComponent implements Radial
 					++symmetricIndex;
 				}
 			}
-			++stageIndex;
-			symmetricIndex = nextStage.getChildCount() - 1;
+			++assemblyIndex;
+			symmetricIndex = searchAssembly.getChildCount() - 1;
 		}
 		return null;
 	}

--- a/core/src/net/sf/openrocket/rocketcomponent/SymmetricComponent.java
+++ b/core/src/net/sf/openrocket/rocketcomponent/SymmetricComponent.java
@@ -570,21 +570,21 @@ public abstract class SymmetricComponent extends BodyComponent implements Radial
 			return null;
 		}
 
-		// might be: (a) Rocket -- for centerline stages
-		//           (b) BodyTube -- for Parallel Stages
-		final AxialStage stage = this.getStage();
-		final RocketComponent stageParent = stage.getParent();
+		final ComponentAssembly assembly = this.getAssembly();
+		// might be: (a) Rocket -- for Centerline/Axial stages
+		//           (b) BodyTube -- for Parallel Stages & PodSets
+		final RocketComponent assemblyParent = assembly.getParent();
 
 		// note:  this is not guaranteed to _contain_ a stage... but that we're _searching_ for one.
-		int stageIndex = stageParent.getChildPosition(stage);
-		int symmetricIndex = this.parent.getChildPosition(this)-1;
+		int assemblyIndex = assemblyParent.getChildPosition(assembly);       // position of stage w/in parent
+		int symmetricIndex = this.parent.getChildPosition(this)-1;  // guess at index of previous stage
 
-		while( 0 <= stageIndex ) {
-			final RocketComponent prevStage = stageParent.getChild(stageIndex);
+		while( 0 <= assemblyIndex ) {
+			final RocketComponent searchAssembly = assemblyParent.getChild(assemblyIndex);
 
-			if(prevStage instanceof AxialStage){
+			if(searchAssembly instanceof ComponentAssembly){
 				while (0 <= symmetricIndex) {
-					final RocketComponent previousSymmetric = prevStage.getChild(symmetricIndex);
+					final RocketComponent previousSymmetric = searchAssembly.getChild(symmetricIndex);
 
 					if (previousSymmetric instanceof SymmetricComponent) {
 						return (SymmetricComponent) previousSymmetric;
@@ -592,8 +592,8 @@ public abstract class SymmetricComponent extends BodyComponent implements Radial
 					--symmetricIndex;
 				}
 			}
-			--stageIndex;
-			symmetricIndex = prevStage.getChildCount() - 1;
+			--assemblyIndex;
+			symmetricIndex = searchAssembly.getChildCount() - 1;
 		}
 		return null;
 	}

